File: javascript/resistor-color/resistor-color.js
export const colorCode = (name = "") => {
  return COLORS.indexOf(name && name.trim().toLowerCase());
};
 
export const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white"
];

File: dart/resistor-color/lib/resistor_color.dart
class ResistorColor {
  final List<String> _colors = [ "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white" ];

  List<String> get colors {
    return _colors;
  }

  int colorCode([final String name = ""]) {
    return colors.indexOf(name.trim().toLowerCase());
  }
}
